.github/workflows/pr.yml

```
@@ -27,21 +48,68 @@ jobs:
 
   lint:
     name: Lint
+    needs: [prepare]
     uses: ./.github/workflows/lint.yml
+    with:
+      mergedSha: ${{ needs.prepare.outputs.mergedSha }}
+      targetSha: ${{ needs.prepare.outputs.targetSha }}
 
   eval:
     name: Eval
+    needs: [prepare]
     uses: ./.github/workflows/eval.yml
     permissions:
       # compare
+      statuses: write
+    secrets:
+      OWNER_APP_PRIVATE_KEY: ${{ secrets.OWNER_APP_PRIVATE_KEY }}
+    with:
+      mergedSha: ${{ needs.prepare.outputs.mergedSha }}
+      targetSha: ${{ needs.prepare.outputs.targetSha }}
+      systems: ${{ needs.prepare.outputs.systems }}
+
+  labels:
+    name: Labels
+    needs: [eval]
+    uses: ./.github/workflows/labels.yml
+    permissions:
       issues: write
       pull-requests: write
-      statuses: write
+
+  reviewers:
+    name: Reviewers
+    needs: [prepare, eval]
+    if: needs.prepare.outputs.targetSha
+    uses: ./.github/workflows/reviewers.yml
     secrets:
       OWNER_APP_PRIVATE_KEY: ${{ secrets.OWNER_APP_PRIVATE_KEY }}
 
   build:
     name: Build
+    needs: [prepare]
     uses: ./.github/workflows/build.yml
     secrets:
       CACHIX_AUTH_TOKEN: ${{ secrets.CACHIX_AUTH_TOKEN }}
+    with:
+      mergedSha: ${{ needs.prepare.outputs.mergedSha }}
+
+  # This job's only purpose is to serve as a target for the "Required Status Checks" branch
+  # protection rule. It depends on all the jobs that are required to pass. If all tests pass, it
+  # doesn't run itself, because "skipped by conditional" counts as "success" in terms of those
+  # branch protection rules.
+  # However, if any of the previous jobs fails, this will run and fail itself as well - and thus
+  # prevent from merging the Pull Request.
+  required:
+    # If you need to add or remove jobs from the list of required status checks, modify this list.
```

```suggestion
    # Modify this list to add or remove jobs from required status checks.
```

or

```suggestion
    # This list defines which jobs are required status checks.
```

or

```suggestion
    # Jobs that are required status checks:
```

or

   
```suggestion
    # The PR cannot merge if any of these fail:
```

or simply:
   
```suggestion
    # Modify this list as required:
```